### PR TITLE
Devtron header 1693

### PR DESCRIPTION
**Pull request by shivani170**
# Description

To make all the headers  in the dashboard uniform components including:
1. Global Configuration
2. Devtron Stack Manager

Screenshot 2022-05-18 at 2 59 43 PM

Fixes https://github.com/devtron-labs/devtron/issues/1693

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Tested in App List
- [x] Tested in Deployment Group
- [x] Tested in Bulk Edits

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/v2/devtronStackManager/DevtronStackManager.component.tsx [hunk 1]
```diff
@@ -227,28 +228,56 @@ export const StackPageHeader = ({
...
+                    headerName="Installed integrations"
+                    onClickCreateButton={handleBreadcrumbClick}
+                    postHeaderName={selectedModule?.title}
+                    isHeaderNameClickable={true}
+                    onClickHeaderName={() => handleRedirectToModule('installed')}
+                />
             )}
-        </section>
+        </>
+        // <section className="page-header bcn-0 border-bottom h-100 flex left">
```
Comment (vivek-devtron) [inline src/components/v2/devtronStackManager/DevtronStackManager.component.tsx:260]: remove commented code

File: src/components/v2/devtronStackManager/DevtronStackManager.component.tsx [hunk 2]
```diff
@@ -227,28 +228,56 @@ export const StackPageHeader = ({
...
+    const handleRedirectToModule = (detailsMode) => {
+        let url =
+            detailsMode === 'discover' ? URLS.STACK_MANAGER_DISCOVER_MODULES : URLS.STACK_MANAGER_INSTALLED_MODULES
+        history.push(url)
+    }
     return (
-        <section className="page-header bcn-0 border-bottom h-100 flex left">
-            {!detailsMode && <div className="flex left page-header__title cn-9 fs-14 fw-6">Devtron Stack Manager</div>}
+        <>
+            {!detailsMode && <PageHeader headerName={'Devtron Stack Manager'} />}
```
[the remark this answers is not in the document]
Comment (shivani170) [reply src/components/v2/devtronStackManager/DevtronStackManager.component.tsx]: yes. done

File: src/components/v2/devtronStackManager/DevtronStackManager.component.tsx (hunk 1 above)
Comment (shivani170) [reply src/components/v2/devtronStackManager/DevtronStackManager.component.tsx]: Done

File: src/components/charts/list/DiscoverCharts.tsx [hunk 3]
```diff
@@ -140,338 +163,474 @@ function DiscoverChartList() {
...
+                    <PageHeader
+                        headerName={state.charts.length === 0 ? 'Chart Store' : 'Deploy multiple charts'}
+                        buttonText="group"
+                        onClickCreateButton={handleToggleChartGroupModal}
+                        showCreateButton={serverMode === SERVER_MODE.FULL && state.charts.length === 0 ? true : false}
+                        CreateButtonIcon={Add}
+                        showIconBeforeText={false}
+                    />
                 </ConditionalWrap>
+                {/* <div className={`page-header `}>
```
Comment (vivek-devtron) [inline src/components/charts/list/DiscoverCharts.tsx:236]: remove commented code

Comment (shivani170) [reply src/components/charts/list/DiscoverCharts.tsx]: Done

File: src/components/charts/list/DiscoverCharts.tsx [hunk 4]
```diff
@@ -140,338 +163,474 @@ function DiscoverChartList() {
...
-                                onClick={(e) => toggleChartGroupModal(!showChartGroupModal)}>
-                            <Add className="icon-dim-18 mr-5" />Create Group
-                        </button>
-                    }
-                </div>
-
-            </div>
-            <Prompt when={isLeavingPageNotAllowed.current} message={'Your changes will be lost. Do you want to leave without deploying?'} />
-            {state.loading || chartListLoading ? <Progressing pageLoader /> : null}
+                    <div className="page-header__cta-container flex">
```
[the remark this answers is not in the document]
Comment (shivani170) [reply src/components/charts/list/DiscoverCharts.tsx:240]: Done

File: src/components/charts/Charts.tsx [hunk 5]
```diff
@@ -1,12 +1,17 @@
-import React from 'react'
+import React, { lazy, useState } from 'react'
```
Comment (vivek-devtron) [inline src/components/charts/Charts.tsx:1]: please remove unused imports

File: src/components/app/details/main.tsx [hunk 6]
```diff
@@ -114,9 +115,6 @@ export function AppHeader() {
         } catch (err) {
             showError(err)
         }
-        return () => {
```
Comment (vivek-devtron) [inline src/components/app/details/main.tsx:117]: is it intentional?

File: src/components/app/list-new/AppList.tsx [hunk 7]
```diff
@@ -597,16 +597,22 @@ export default function AppList() {
...
+    }
+
     function renderPageHeader() {
         return (
             <Fragment>
                 <PageHeader
                     headerName="Applications"
-                    buttonText="new"
-                    onClickCreateButton={handleCreateButton}
                     showCreateButton={serverMode === SERVER_MODE.FULL ? true : false}
```
Comment (vivek-devtron) [inline src/components/app/list-new/AppList.tsx:614]: can we add this check in renderCreateButton and remove from here?

File: src/components/common/header/PageHeader.tsx [hunk 8]
```diff
@@ -1,47 +1,53 @@
 import React from 'react'
 import Tippy from '@tippyjs/react'
+import { ReactComponent as Close } from '../../../assets/icons/ic-close.svg'
+
 export interface PageHeaderType {
     headerName?: string
-    buttonText?: string
     onClickCreateButton?: () => void
```
Comment (vivek-devtron) [inline src/components/common/header/PageHeader.tsx:7]: can we remove onClickCreateButton,  showCreateButton?

File: src/components/deploymentGroups/DeploymentGroupList.tsx [hunk 9]
```diff
@@ -303,17 +303,19 @@ export default class DeploymentGroupList extends Component<BulkActionListProps,
...
             <div>
-                <PageHeader
-                    headerName="Deployment Groups"
-                    showCreateButton={this.state.view === ViewType.FORM ? true : false}
-                    buttonText="group"
-                    onClickCreateButton={this.redirectToCreateGroup}
-                    CreateButtonIcon={Add}
-                    showIconBeforeText={true}
-                />
+                <PageHeader headerName="Deployment Groups" />
```
Comment (vivek-devtron) [inline src/components/deploymentGroups/DeploymentGroupList.tsx:318]: did we forget to add renderCreateButton here?

File: src/components/v2/headers/ChartHeader.component.tsx [hunk 10]
```diff
@@ -135,7 +135,7 @@ function ChartHeaderComponent() {
     }
 
     return (
-        <div className="app-details-page">
+        <div className="helm-app-page-header">
```
Comment (vivek-devtron) [inline src/components/v2/headers/ChartHeader.component.tsx:138]: this class change is intentional?

Comment (shivani170) [reply src/components/v2/headers/ChartHeader.component.tsx:138]: Yes

File: src/components/deploymentGroups/DeploymentGroupList.tsx (hunk 9 above)
Comment (shivani170) [reply src/components/deploymentGroups/DeploymentGroupList.tsx]: fixed

File: src/components/app/details/main.tsx (hunk 6 above)
Comment (shivani170) [reply src/components/app/details/main.tsx:117]: Yes i was adding some hack to render. but forget to remove earlier

File: src/components/app/list-new/AppList.tsx (hunk 7 above)
Comment (shivani170) [reply src/components/app/list-new/AppList.tsx]: done

File: src/components/common/header/PageHeader.tsx (hunk 8 above)
Comment (shivani170) [reply src/components/common/header/PageHeader.tsx]: Yes done

File: src/components/common/header/PageHeader.tsx [hunk 11]
```diff
@@ -69,7 +66,7 @@ function PageHeader({
                 )}
             </h1>
             {showTabs && renderHeaderTabs()}
-            {showCreateButton && renderCreateButton()}
+            {renderCreateButton && renderCreateButton()}
```
Comment (vivek-devtron) [inline src/components/common/header/PageHeader.tsx:69]: Can we add functionType check here rather than renderCreateButton this?

Comment (vivek-devtron) [reply src/components/common/header/PageHeader.tsx:69]: We should also rename this from renderCreateButton to renderCTAs or renderActionButtons may be. or something similar

**Review by vivek-devtron — APPROVED** [review]

**Review by pghildiyal — APPROVED** [review]

**Pull request merged**